ext/opcache/jit/zend_jit_trace.c

```diff
@@ -8585,7 +8585,7 @@ int ZEND_FASTCALL zend_jit_trace_exit(uint32_t exit_num, zend_jit_registers_buf
 				if (op->opcode == ZEND_FETCH_DIM_IS || op->opcode == ZEND_FETCH_OBJ_IS) {
 					ZVAL_NULL(EX_VAR_NUM(i));
 				} else {
-					assert(op->opcode == ZEND_FETCH_DIM_R || op->opcode == ZEND_FETCH_LIST_R || op->opcode == ZEND_FETCH_OBJ_R);
+					assert(op->opcode == ZEND_FETCH_DIM_R || op->opcode == ZEND_FETCH_LIST_R || op->opcode == ZEND_FETCH_OBJ_R || op->opcode == ZEND_FETCH_DIM_FUNC_ARG);
```

Comment:
ZEND_FETCH_OBJ_FUNC_ARG should be add as well.

```php
<?php
namespace Foo;
class X {
        public $a = 1;
        public $b;
        function __construct() {
                unset($this->b);
        }
}
function test() {
    $a['x'] = new X;
    for ($fusion = 0; $i < 3; $i++) {
        var_dump($a['x']->b);
    }
}
test();
?>
```

Also``assert()`` may be changed to ``ZEND_ASSERT()``.

Reply:
Right, makes sense, thanks